src/Symfony/Component/HttpKernel/DependencyInjection/ResettableServicePass.php

```diff
@@ -56,14 +56,9 @@ public function process(ContainerBuilder $container)
             $methods[$id] = $attributes['method'];
         }
 
-        if (empty($services)) {
-            $container->removeDefinition(ServiceResetListener::class);
-
-            return;
+        if (!empty($services)) {
+            $container->register('request_stack', ResettingRequestStack::class)
```

Comment:
Shouldn't we decorate the existing request stack rather than replacing it?

Reply:
to achieve what?

Reply:
Someone might already have replaced the request stack by its own, or decorated it for its own needs.

Reply:
OK, indeed. So I'm going to add a runtime decorating proxy.

Reply:
Wait no! that's not how decoration works! So there is no issue with it here.
We don't guarantee BC for ppl that just replace services (otherwise we would be is serious trouble).
So as is is all good, isn't it?

Reply:
A `request_stack` decorator would just end up with a ResettingRequestStack instance as inner since this pass runs before decoration happens. Seems fine

Reply:
Alright